### PR TITLE
Relay errors from Rust to Python

### PR DESCRIPTION
**Pull request by entwanne**
We now have some error handling instead of panics and/or segmentation faults.
It has to be improved to add different error types.

Next step: split & clean the code.

### CODE REVIEW

File: slonik/rust.py [hunk 1]
```diff
@@ -50,44 +50,33 @@ def __del__(self):
...
-    # get collected before the string is used.
-    attached_refs[rv] = s
-    return rv
+    result = func(*args)
+    if isresult(result):
+        if result.status:
+            error = ffi.cast('_Error*', result.data)
+            try:
+                error_msg = buff_to_bytes(lib.error_msg(error)).decode()
+                raise Exception(error_msg)
```
Comment (k4nar) [inline slonik/rust.py]: We should probably return a custom exception (and we'll add specialized errors at some point as well).

File: tests/test_errors.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+def test_query_error(conn):
+    with pytest.raises(Exception) as e:
+        list(conn.query('SELECT bar FROM foo'))
+    assert 'relation "foo" does not exist' in str(e)
+
+
+def test_execute_error(conn):
+    with pytest.raises(Exception) as e:
+        conn.execute('DELETE FROM foo')
+    assert 'relation "foo" does not exist' in str(e)
```
Comment (k4nar) [inline tests/test_errors.py:27]: Could you add a tests with a syntax error?

File: slonik/rust.py (hunk 1 above)
Comment (entwanne) [reply slonik/rust.py]: Yes, the idea is to have in the future a base error for slonik exceptions and a hierarchy based on the status code.

**Comment by entwanne** [comment]
@k4nar UP

**Review by k4nar — APPROVED** [review]

**Pull request merged**